cmd/backup/main.go

```diff
@@ -65,4 +82,83 @@ func main() {
 	s.must(s.withLabeledCommands(lifecyclePhaseProcess, s.encryptArchive)())
 	s.must(s.withLabeledCommands(lifecyclePhaseCopy, s.copyArchive)())
 	s.must(s.withLabeledCommands(lifecyclePhasePrune, s.pruneBackups)())
+
+	return nil
+}
+
+func main() {
+	serve := flag.Bool("foreground", false, "run the backup in the foreground")
+	envFolder := flag.String("env-folder", "/etc/dockervolumebackup/conf.d", "location of environment files")
+	flag.Parse()
+
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+
+	if *serve {
+		logger.Info("running in the foreground instead of cron")
+
+		cr := cron.New()
+
+		addJob := func(c *Config) {
+			logger.Info("added cron job", "schedule", c.BackupCronExpression)
+			_, err := cr.AddFunc(c.BackupCronExpression, func() {
+				err := runBackup(c)
+				if err != nil {
+					logger.Error("unexpected error during backup", "error", err)
+				}
+			})
+			if err != nil {
+				logger.Error("failed to create cron job", "schedule", c.BackupCronExpression)
+			}
+		}
+
+		cs, err := loadEnvFiles(*envFolder)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				logger.Error("could not load config from environment files")
+				os.Exit(1)
+			}
+
+			c, err := loadEnvVars()
+			if err != nil {
+				logger.Error("could not load config from environment variables")
+				os.Exit(1)
+			} else {
+				addJob(c)
+			}
+		} else {
+			for _, c := range cs {
+				addJob(c)
+			}
+		}
+
+		logger.Info("subscribed to interupt signals")
+		var quit = make(chan os.Signal, 1)
+		signal.Notify(quit, syscall.SIGTERM, syscall.SIGINT)
+
+		logger.Info("starting cron scheduler")
+		cr.Start()
+
+		logger.Info("application goes to sleep")
+		<-quit
+
+		logger.Info("interrupt arrived, stopping schedules")
+		ctx := cr.Stop()
+		<-ctx.Done()
+	} else {
+		logger.Info("executing one time backup")
+
+		c, err := loadEnvVars()
+		if err != nil {
+			logger.Info("could not load config from environment variables", "error", err)
+			os.Exit(1)
+		}
+
+		err = runBackup(c)
+		if err != nil {
+			logger.Info("unexpected error during backup", "error", err)
+			os.Exit(1)
+		}
+	}
+
+	os.Exit(0)
```

Comment:
I'd remove this as:
- it's the default behavior when there's nothing left to do for the program
- this can sneakily terminate the program while we still have dangling goroutines or similar running. Without force exiting, we'll notice such things quicker

Reply:
Done